core/shared/src/test/scala/fs2/AsyncPropertySuite.scala

```diff
@@ -0,0 +1,146 @@
+package fs2
+
+import scala.concurrent.Future
+
+import cats.data.State
+import cats.effect.Effect
+import cats.effect.implicits._
+import cats.implicits._
+
+import munit.{Location, ScalaCheckSuite}
+
+import org.scalacheck.{Arbitrary, Gen}
+import org.scalacheck.rng.Seed
+
+trait AsyncPropertySupport extends ScalaCheckSuite {
+
+  private def samples[X](
+      g: Gen[X],
+      params: Gen.Parameters = Gen.Parameters.default
+  ): State[Seed, List[X]] = {
+    def go(acc: List[X], rem: Int, seed: Seed): (Seed, List[X]) =
+      if (rem <= 0) (seed, acc.reverse)
+      else {
+        val r = g.doPureApply(params, seed)
+        r.retrieve match {
+          case Some(x) => go(x :: acc, rem - 1, r.seed)
+          case None    => go(acc, rem, r.seed)
+        }
+      }
+    State(s => go(Nil, scalaCheckTestParameters.minSuccessfulTests, s))
+  }
+
+  private def reportPropertyFailure[F[_]](f: F[Unit], seed: Seed, describe: => String)(implicit
+      F: Effect[F],
+      loc: Location
+  ): F[Unit] =
+    f.handleErrorWith { t =>
+      fail(s"Property failed with seed ${seed.toBase64} and params: " + describe, t)
+    }
+
+  def forAllAsync[F[_], A](
+      f: A => F[Unit]
+  )(implicit arbA: Arbitrary[A], F: Effect[F], loc: Location): Future[Unit] = {
+    val seed = Seed.random()
+    samples(arbA.arbitrary)
+      .runA(seed)
+      .value
+      .traverse_(a => reportPropertyFailure(f(a), seed, a.toString))
+      .toIO
+      .unsafeToFuture
```

Comment:
Curious: can these `forAllAsync` methods return `F[Unit]` or just `IO[Unit]`? Seems possible since we have the `munitValueTransform` for `IO ~> Future` but I might be missing something

Reply:
Good catch! I changed these to return `F[Unit]` which means the constraint goes from `Effect[F]` to `ApplicativeError[F, Throwable]`. I added a `SyncIO` runner to `Fs2Suite` as well.